cms/urls.py

```diff
@@ -144,60 +160,59 @@
 }
 
 if settings.FEATURES.get('ENABLE_CONTENT_LIBRARIES'):
-    urlpatterns += (
+    urlpatterns += [
         url(r'^library/{}?$'.format(LIBRARY_KEY_PATTERN),
-            'contentstore.views.library_handler', name='library_handler'),
+            contentstore.views.library_handler, name='library_handler'),
         url(r'^library/{}/team/$'.format(LIBRARY_KEY_PATTERN),
-            'contentstore.views.manage_library_users', name='manage_library_users'),
-    )
+            contentstore.views.manage_library_users, name='manage_library_users'),
+    ]
 
 if settings.FEATURES.get('ENABLE_EXPORT_GIT'):
-    urlpatterns += (url(
-        r'^export_git/{}$'.format(
-            settings.COURSE_KEY_PATTERN,
-        ),
-        'contentstore.views.export_git',
-        name='export_git',
-    ),)
+    urlpatterns += [
+        url(r'^export_git/{}$'.format(settings.COURSE_KEY_PATTERN),
+            contentstore.views.export_git,
+            name='export_git')
+    ]
 
 if settings.FEATURES.get('ENABLE_SERVICE_STATUS'):
-    urlpatterns += patterns(
-        '',
-        url(r'^status/', include('openedx.core.djangoapps.service_status.urls')),
-    )
+    urlpatterns.append(url(r'^status/', include('openedx.core.djangoapps.service_status.urls')))
 
 if settings.FEATURES.get('AUTH_USE_CAS'):
-    urlpatterns += (
-        url(r'^cas-auth/login/$', 'openedx.core.djangoapps.external_auth.views.cas_login', name="cas-login"),
-        url(r'^cas-auth/logout/$', 'django_cas.views.logout', {'next_page': '/'}, name="cas-logout"),
-    )
+    urlpatterns += [
+        url(r'^cas-auth/login/$', openedx.core.djangoapps.external_auth.views.cas_login, name="cas-login"),
+        url(r'^cas-auth/logout/$', django_cas.views.logout, {'next_page': '/'}, name="cas-logout"),
+    ]
 
-urlpatterns += patterns('', url(r'^admin/', include(admin.site.urls)),)
+urlpatterns.append(url(r'^admin/', include(admin.site.urls)))
 
 # enable entrance exams
 if settings.FEATURES.get('ENTRANCE_EXAMS'):
-    urlpatterns += (
-        url(r'^course/{}/entrance_exam/?$'.format(settings.COURSE_KEY_PATTERN), 'contentstore.views.entrance_exam'),
-    )
+    urlpatterns.append(url(r'^course/{}/entrance_exam/?$'.format(settings.COURSE_KEY_PATTERN),
+                       contentstore.views.entrance_exam))
 
 # Enable Web/HTML Certificates
 if settings.FEATURES.get('CERTIFICATES_HTML_VIEW'):
-    urlpatterns += (
+    from contentstore.views.certificates import (
+        certificate_activation_handler,
```

Comment:
These caused problems if imported at the top of the module?

Reply:
It was a conservative choice because certificates is one of only a couple of views files that are not import *'d into cms/djangoapps/contentstore/views/__init__.py. I took that to mean that it was undesirable to import it if the setting was off. It's a pattern used in a couple of other places in the file, so it seemed sane.